sc/src/test/java/uk/ac/ebi/atlas/download/FileDownloadControllerWIT.java

```diff
@@ -122,4 +126,25 @@ void downloadArchiveForInvalidExperimentFileType() throws Exception {
                 .andExpect(status().is(404))
                 .andExpect(view().name("error-page"));
     }
+
+    @Test
+    void downloadArchiveForValidExperimentAccessionList() throws Exception {
+        String expectedArchiveName =
+                MessageFormat.format(
+                        ARCHIVE_NAME, EXPERIMENT_ACCESSION_LIST.size(), "experiments");
+        this.mockMvc.perform(get(ARCHIVE_DOWNLOAD_LIST_URL)
+                .param("experimentAccessions",  EXPERIMENT_ACCESSION_LIST.toString().replace("[","").replace("]", "")))
+                .andExpect(status().isOk())
+                .andExpect(
+                        header().string(HttpHeaders.CONTENT_DISPOSITION, "attachment;filename=" + expectedArchiveName))
+                .andExpect(content().contentType("application/zip"));
+    }
+
+    @Test
+    void downloadArchiveForInvalidExperimentAccessionList() throws Exception {
```

Comment:
What happens if I enter a mixed list of valid and invalid experiment accessions? Probably not what you’d expect!

Reply:
Good point! I have added an `if` condition in the controller and a test for a mix of valid and invalid experiment accessions.